tests/unit/integrations/bitbucket/test_bitbucket_repos.py

```diff
@@ -0,0 +1,391 @@
+"""Tests for Bitbucket repository service URL parsing."""
```

Comment:
@openhands lets simplify this test file

1. let's only add tests for `search_repository` for now where we're correctly parsing the URL
2. remove all the other tests from here
3. check the parameters during the API call and verify that the right workspace repo combination was included in the call